apps/scopes/test/e2e/run-native-tests.sh

```
@@ -0,0 +1,216 @@
+echo "Binary: $BINARY_PATH"
+echo "========================================="
+
+TOTAL_TESTS=0
+PASSED_TESTS=0
+FAILED_TESTS=0
+
+# Function to run a test
+run_test() {
+    local test_name="$1"
+    shift
+    local command_args="$@"
```

Shellcheck (0.10.0)

[warning] Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.

(SC2124)